tests/robot-tests/tests/libs/utilities.py

```diff
@@ -87,8 +87,20 @@ def raise_assertion_error(err_msg):
 
 
 def user_waits_until_parent_contains_element(
-    parent_locator: object, child_locator: str, timeout: int = None, error: str = None, count: int = None
+    parent_locator: object, child_locator: str, timeout: int = None, error: str = None, count: int = None, retries: int = 5, delay: float = 1.0
 ):
+    def _try_with_retries_child_element_found(func, *args, **kwargs):
```

Comment:
Assuming that this sort of nesting is the equivalent of a `private` function, i.e. only accessible within the parent function scope, I think this would be better positioned _outside_ of the parent function as it appears to be quite generic, and could be used in other places where retries are needed.

Also I think the function's name would benefit from being a bit clearer, e.g. `retry_or_fail_with_delay`, or something similarly generic.